src/main/java/com/emc/pravega/perf/PravegaPerfTest.java

```diff
@@ -259,131 +314,278 @@ private static void parseCmdLine(String[] args) {
     }
 
-    /**
-     * A Sensor simulator class that generates dummy value as temperature measurement and ingests to specified stream.
-     */
 
-    private static class WriterWorker implements Runnable {
+    private static class StartLocalService {
+        static final int PORT = 9090;
+        static final String SCOPE = "Scope";
+        static final String STREAM_NAME = "aaj";
+    }
+}
+
+class StreamHandler{
+      String scope;
+      String stream;
+      String controllerUri;   
+      ControllerImpl controller; 
+      StreamManager streamManager;
+      StreamConfiguration streamconfig;
+      ReaderGroupManager readerGroupManager; 
+      ReaderGroup readerGroup;
+      ScheduledExecutorService bgexecutor;
+      int segCount;
+      int timeout;
+
+
+      StreamHandler(String scope, String stream, 
+                    String uri, int segs,
+                    int timeout,  ControllerImpl contrl,
+                    ScheduledExecutorService bgexecutor) throws Exception {
+               this.scope = scope;
+               this.stream = stream;
+               this.controllerUri= uri;
+               this.controller = contrl;
+               this.segCount = segs; 
+               this.timeout = timeout;
+               this.readerGroup = null;
+               this.bgexecutor = bgexecutor;
+               streamManager = StreamManager.create(new URI(uri));
+               streamManager.createScope(scope);
+               streamconfig = null;
+
+       } 
+
+
+       boolean create() throws Exception {
+               if (streamconfig == null)
+                   streamconfig = StreamConfiguration.builder().scope(scope).streamName(stream)
+                                                 .scalingPolicy(ScalingPolicy.fixed(segCount))
+                                                 .build();
+   
+               return streamManager.createStream(scope, stream,streamconfig);
+       }  
+
+       boolean scale() throws Exception {
+               StreamSegments segments = controller.getCurrentSegments(scope, stream).join();
+               final int nseg = segments.getSegments().size();
+               System.out.println("Current segments of the stream: "+stream+ " = " + nseg);
+                
+               if (nseg == segCount) {
+                  return true;  
+               }
+                  
+               System.out.println("The stream: " + stream + " will be manually scaling to "+ segCount+ " segments");
+
+               /*
+                * Note that the Upgrade stream API does not change the number of segments;
+                * but it indicates with new number of segments.
+                * after calling update stream , manual scaling is required
+                */
+               if (!streamManager.updateStream(scope, stream, streamconfig)) {
+                    System.out.println("Could not able to update the stream: " + stream + " try with another stream Name");
+                    return false;   
```

Comment:
Boolean error codes is strongly discouraged in Java. 
This should be an exception of an appropriate type with the string from the line above.